bob/src/cluster/operations.rs

```diff
@@ -74,10 +86,30 @@ pub(crate) async fn put_at_least(
     at_least: usize,
     options: PutOptions,
 ) -> (Tasks, Vec<NodeOutput<Error>>) {
-    let mut handles: FuturesUnordered<_> = target_nodes
-        .cloned()
-        .map(|node| call_node_put(key, data.clone(), node, options.clone()))
-        .collect();
+    call_at_least(target_nodes, at_least, |n| {
+        call_node_put(key, data.clone(), n, options.clone())
+    })
+    .await
+}
+
+pub(crate) async fn delete_at_least(
```

Comment:
`delete` operation is a little bit special. It should operate on every node and success only if the `delete` on every node was successful. Thus `at_least` parameter is not needed, nor is `at_least` in the function name

Reply:
Since it'll use `call_at_least` internally, I'll leave `at_least` suffix here and pass count of target nodes on the call site